nemo-physical/src/function/definitions/string.rs

```diff
@@ -11,6 +12,17 @@ use super::{
     BinaryFunction, FunctionTypePropagation, NaryFunction, TernaryFunction, UnaryFunction,
 };
 
+/// Unicode friendly version of [String] find
+///
+/// Returns the index of the first occurrence of the needle in the haystack
+/// or `None` if the needle is not found.
+fn unicode_find(haystack: &String, needle: &String) -> Option<usize> {
+    let haystack_graphemes = haystack.graphemes(true).collect::<Vec<&str>>();
+    let needle_graphemes = needle.graphemes(true).collect::<Vec<&str>>();
+    (0..(haystack_graphemes.len() - needle_graphemes.len() + 1))
```

Comment:
The subtraction can underflow if `needle` is longer than `haystack`. Better use `saturating_sub` here.

Reply:
Very nice catch! I think the solution is just doing a length check within the function.

If `needle` is longer than `haystack` and we do a `saturating_sub` (neat I did not know this function existed), we'll immediately evaluate `haystack_graphemes[i..i + needle_graphemes.len()]` which will result in an error since `needle` is longer than `haystack`.